test/index.js

```diff
@@ -0,0 +1,37 @@
+var scrape = require("readable-proxy").scrape;
+var path = require("path");
+var fs = require("fs");
+var prettyPrint = require("html").prettyPrint;
+var chai = require("chai");
+var chaiAsPromised = require("chai-as-promised");
+chai.should();
+chai.use(chaiAsPromised);
+var expect = chai.expect;
+
+var testPageRoot = path.join(__dirname, "test-pages");
+var testPages = fs.readdirSync(testPageRoot).map(function(dir) {
+  return {
+    dir: dir,
+    source: path.join(testPageRoot, dir, "source.html"),
+    expected: path.join(testPageRoot, dir, "expected.html"),
+  };
+});
+
+describe("Test page", function() {
+  testPages.forEach(function(testPage) {
+    describe(testPage.dir, function() {
+      it("should render as expected", function() {
+        // Allows up to 10 seconds for parsing to complete.
+        // XXX: Scraping is damn slow. Investigate.
+        this.timeout(10000);
```

Comment:
To be honest that was the shortest plan I could find to move fast, as most of required code was already available in readable-proxy.

Also don't forget we need a real DOM to run Readability, and node doesn't ship with one. So while I could possibly fire jsdom, karma, selenium (!) and so on, I found easier to reuse what's been made available by the proxy.

To me the main issue is the quality of Readability itself, both from an API PoV (which is debatable) and an accuracy one. Actually I'm finding more and more case where it behaves completely wrong, eg. stripping or *reordering* large portions of text, eg.:

![screen shot 2015-03-14 at 21 37 38](https://cloud.githubusercontent.com/assets/41547/6653415/62a5021c-ca92-11e4-9495-898a5c5f821a.png)

This is like voodoo to me, the most worrying part is that after a few days trying to figure out how the lib works precisely, I'm still struggling. That's no good sign.

Anyway, I'll try to come with a better testing infrastructure on Monday, for now I'll try enjoying my week-end :p